### PR TITLE
Incubator.Dialog - header refactor

### PR DESCRIPTION
**Pull request by M-i-k-e-l**
## Description
Incubator.Dialog - header refactor

## Changelog
Incubator.Dialog - header refactor

### CODE REVIEW

File: src/incubator/Dialog/DialogHeader.tsx [hunk 1]
```diff
@@ -1,30 +1,42 @@
...
 const DialogHeader = (props: DialogHeaderProps = {}) => {
   const {text = {}, renderContent, showKnob = true, showDivider = true, ...others} = props;
 
+  const knob = useMemo(() => {
+    if (showKnob) {
+      return <View style={styles.knob}/>;
+    }
+  }, [showKnob]);
+
   const _renderContent = useMemo(() => {
```
Comment (Inbal-Tish) [inline src/incubator/Dialog/DialogHeader.tsx:19]: I would change "_renderContent" to "headerContent" to be extra clear and follow the other component's naming

Comment (M-i-k-e-l) [reply src/incubator/Dialog/DialogHeader.tsx]: Missed that, 10x
Maybe I should rename the prop as well 🤔

Comment (M-i-k-e-l) [reply src/incubator/Dialog/DialogHeader.tsx]: But it's under `headerProps`, WDYT?

Comment (Inbal-Tish) [reply src/incubator/Dialog/DialogHeader.tsx]: I think the prop name is good

**Review by Inbal-Tish — APPROVED** [review]

**Pull request merged**